daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/Parser.scala

```diff
@@ -190,6 +190,8 @@ class ChoiceParser(
     
     var successfullyParsedChildBranch = false
 
+    val savedLastChildNode = pstate.infoset.contents.lastOption
```

Comment:
Probably good to declare types on public val/var. In this case we're using lastOption, so an Option type, which is not usual/preferred in the runtime.

Reply:
Will fix.

Reply:
Actually, this isn't a public member, it's just a local val. Do you think the type is still needed in this case?